### PR TITLE
more fields and name consistency

### PR DESCRIPTION
**Pull request by HailSanta**

### CODE REVIEW

File: include/enums.h [hunk 1]
```diff
@@ -583,8 +583,8 @@ enum SoundIDs {
     SOUND_376                       = 0x00000376,
     SOUND_377                       = 0x00000377,
     SOUND_379                       = 0x00000379,
-    SOUND_37A                       = 0x0000037A,
-    SOUND_ELECTRIC_DAMAGE           = 0x0000037B,
+    SOUND_HIT_PLAYER_SHOCK                       = 0x0000037A,
+    SOUND_HIT_SHOCK           = 0x0000037B,
```
Comment (Brotenko) [inline include/enums.h:587]: Not really an issue, but fixing the spacing would be neat

**Review by Brotenko — APPROVED** [review]
Looks good to me tbh besides the one "beauty error" in the enum

File: include/common_structs.h [hunk 2]
```diff
@@ -2118,16 +2120,23 @@ typedef struct {
...
-    /* 0x10 */ char unk_10[0x20];
-    /* 0x30 */ s32 unk_30;
-} Temp8010F250; // size = 0x34
+typedef struct PlayerSpinState {
+    /* 0x00 */ s8 stopSoundTimer;
+    /* 0x01 */ s8 hasBufferedSpin;
+    /* 0x02 */ s8 hitWallTime; // incremented while blocked by a wall
+    /* 0x03 */ s8 spinCountdown;
+    /* 0x04 */ s32 prevActionState;
+    /* 0x08 */ s32 bufferedStickAxis[2];
```
Comment (nanaian) [inline include/common_structs.h:2129]: Vec2s?

File: include/common_structs.h [hunk 3]
```diff
@@ -2118,16 +2120,23 @@ typedef struct {
...
+    /* 0x10 */ f32 spinDirectionMagnitude;
+    /* 0x14 */ f32 spinDirection[2];
```
Comment (nanaian) [inline include/common_structs.h:2131]: Vec2f?

File: include/enums.h [hunk 4]
```diff
@@ -1162,6 +1162,13 @@ enum Phases {
     PHASE_PLAYER_HAPPY              = 30,
 };
 
+enum ActorClasses {
+    ACTOR_CLASS_PLAYER      = 0x000,
+    ACTOR_CLASS_PARTNER     = 0x100,
+    ACTOR_CLASS_ENEMY       = 0x200,
+    ACTOR_CLASS_MASK        = 0x700,
+};
```
Comment (nanaian) [inline include/enums.h:1170]: Can you remove the enum with ENEMY0 etc

File: include/enums.h [hunk 5]
```diff
@@ -1402,6 +1409,7 @@ enum ActionStates {
     ACTION_STATE_IDLE                           = 0x00000000,
     ACTION_STATE_WALK                           = 0x00000001,
     ACTION_STATE_RUN                            = 0x00000002,
+    // all states above are considered locomotion states
```
Comment (nanaian) [inline include/enums.h:1412]: By which func(s)? Would be good to reference it

**Review by nanaian — CHANGES_REQUESTED** [review]

File: src/actor_api.c [hunk 6]
```diff
@@ -822,20 +822,20 @@ ApiStatus SetAnimation(Evt* script, s32 isInitialCall) {
 ApiStatus GetAnimation(Evt* script, s32 isInitialCall) {
     Bytecode* args = script->ptrReadPos;
     s32 actorID = evt_get_variable(script, *args++);
-    s32 var1;
+    s32 partID;
     ActorPart* actorPart;
-    s32 a1;
+    s32 outVat;
```
Comment (JoshDuMan) [inline src/actor_api.c:827]: outVat -> outVar

**Review by JoshDuMan — COMMENTED** [review]
Good PR 👍

File: include/enums.h (hunk 5 above)
Comment (HailSanta) [reply include/enums.h:1412]: ill add them as i find them. its pretty common in the player action states / physics functions iirc.

File: include/enums.h (hunk 4 above)
Comment (HailSanta) [reply include/enums.h:1170]: we still need the other enum to refer to actors by their IDs. SELF is especially important, but you'll find battle sections (mostly bosses) with AI tied to hard-coded actor IDs.

**Review by JoshDuMan — APPROVED** [review]

**Review by ethteck — APPROVED** [review]
Thanks so much for this. You're doing cod's work.

**Pull request merged**